### DISCUSSION TITLE
O que falta nas pessoas que vocês entrevistam?

### DISCUSSION THREAD

**Discussion by woliveiras** [frontendbr/forum#911] [Discussão]
Olá pessoal!

Como parte do nosso trabalho, as vezes, entrevistamos pessoas para nossas equipes.

Os requisitos, técnicos ou não, para as vagas, normalmente, são acordados entre a equipe para suprir necessidades específicas do grupo, da empresa, do produto, etc.

Normalmente entrevistamos uma quantidade legal de pessoas antes de fechar uma contratação.

Gostaria de saber de vocês, que também entrevistam, **quais os skills que estão faltando nas pessoas que estão participando dos seus processos seletivos?**

Seria legal listar tudo mesmo, como frameworks, linguagens, paradigmas, sistema de controle de versão, linguagem, habilidade de comunicação, auto gestão, resiliência, etc., etc.

A ideia dessa issue é levantar o máximo de dados sobre o que está faltando nas pessoas mesmo. Por isso podem colocar coisas repetidas ou só reforçar o que alguém comentou.

Lembrando que o intuito aqui é levantar dados sobre participantes dos processos seletivos, não reclamar sobre os processos, sobre as empresas ou coisa do tipo, ein. :joy:

**Comment by vinicius73** [comment]
Dente vários pontos, o que mais dificulta, é a __falta de atualização__ dos candidatos.

Não digo que eles deveriam estar utilizando as __últimas modas__, porém saber quais são é extremamente saudável.

Ex: ainda há muito __dev__ que não viu nada sobre ES2015+, muitos nem sabem doq se trata.

**Comment by bernarddeluna** [comment]
Conversamos com mais de 150 programadores por semana para oportunidades. Vou tentar separar em hard skills e soft skills ok?

## Hard Skills
- React está sendo muito pedido ainda, não vimos ainda demanda de VUE no mercado (embora eu acredite que vai aumentar esse ano).
- Angular não é demandado quase, porém ainda tem muito dev que não parou para estudar react e se permitir inclusive aumentar o valor de mercado.
- Javascript é demandado para QUALQUER oportunidade, inclusive em vagas Back-end. Muitos Backs mais velhos não conseguem mais continuar no hype das oportunidades por não terem evoluído seu lado no Javascript
- Muitos devs de Agencia não investiram/investem em sair do jQuery, aprendendo mais js nativo, ou até evoluindo para uma das tecnologias faladas anteriormente.
- Precisa saber ES6 para ser considerado top

## Soft Skills
- Não brilho no olho para resolver problemas da empresa, se colocando bem limitado na função
- Relaciona bastante a oportunidade em suas comunicações a salario, não pensando no fator "5 anos"
- Pontualidade é um problema sério
- Não sabe se comunicar. Ou só responde o que é perguntado, mas não evolui, não provoca, nem mesmo mostra excitação com as possibilidades. Aqueles que são diferentes ganham muitos pontos, inclusive passam na frente muitas vezes dos que são "melhores" tecnicamente.


Uma coisa que eu acho que vale a pena mencionar é a formação. Muitos devs não consideram relevante ter uma graduação. Não é que seja obrigatório e que essa pessoa não vá conseguir dar certo sem a faculdade. Mas ela abre mão de algumas oportunidades muito boas por conta disso.

Ou seja, se eu pudesse dar uma dica, gradue-se, seu valor de mercado aumentará, gerando mais possibilidades para empresas aqui no Brasil e PRINCIPALMENTE oportunidades fora do país.

**Comment by diessica** [comment]
> [...] Ou só responde o que é perguntado, mas não evolui, não provoca, nem mesmo mostra excitação com as possibilidades. Aqueles que são diferentes ganham muitos pontos, inclusive passam na frente muitas vezes dos que são "melhores" tecnicamente.

esse é o principal ponto para mim, e descreve a maioria das pessoas que trabalhei na minha vida, infelizmente. é muito mais fácil enfiar es6 na cabeça de alguém do que desenvolver na pessoa algo como senso crítico, que altera o jeito que ela pensa e processa as coisas.

comunicação também é mais fácil de se desenvolver. o problema mesmo é *o que* a pessoa comunica, ou se ela *só* comunica – não muda nada, não quer ser protagonista, só fica ali sugando e superestimando quem faz.

**Comment by klaygomes** [comment]
Além dos acima citados, algo que acontece muito e incomoda é a falta de interesse pelo cargo/empresa que esta se candidatando. Já entrevistei pessoas que pareciam não fazer ideia do que estavam fazendo ali (sério!). 

Procurar entender o negócio da empresa e se mostrar um pouco interessado na empresa já faz uma puta diferença.

**Comment by Tautorn** [comment]
Além do que citaram acima, o que pecam muito é a análise. Muitos devs estão focados apenas em programar, é não visualizam o motivo daquilo tudo. 
Escrever código é "fácil", o difícil é agregar valor com a entrega.
Como citou o @klaygomes, conhecer o negócio da empresa faz toda a diferença. 
Tecnologia é "apenas" ferramenta.

**Comment by bernardodiasc** [comment]
> Muitos devs estão focados apenas em programar

Não são devs, são apenas programadores 😄. Desenvolvedor também programa, mas programador não necessariamente desenvolve...

**Comment by Tautorn** [comment]
> Não são devs, são apenas programadores smile. Desenvolvedor também programa, mas programador não necessariamente desenvolve...

Com certeza @bernardodiasc, mas é um problema muito comum, até mesmo por pessoas com "experiência", tidos como desenvolvedores.

**Comment by adaltojunior86** [comment]
Eu estou a pouco tempo como team-leader e passando a fazer parte do processo de contratação. A grande maioria não atende os hards skills. Após a primeira entrevista nós enviamos um desafio pra ser respondido, mas quase não temos retorno, ou o desafio vem incompleto então não passamos pra etapa seguinte.

**Comment by vedovelli** [comment]
Sou tech lead e desenvolvedor full stack na www.octimine.com.

Nesta posição já entrevistei mais de uma dezena de candidatos.

O que mais me deixou consternado foi que, apesar de todos terem um perfil no Github, a grande maioria não tinha qualquer código. Nenhum repositório, nenhum fork, nada.

Eu consigo bastante difícil realmente estudar algo sem desenvolver algum pet project.

**Comment by dtelaroli** [comment]
Dando aula em curso técnico de tecnologia, percebi que os estudantes não sabem o básico de matemática o que impossibilita um bom aproveitamento do curso. Um outro ponto fundamental é o pouco ou nenhum interesse em estudar e sabemos que em nossa área quem não estuda não consegue ter um bom desempenho profissional. 

Tratando de candidatos às vagas o que tenho visto são graduados que não tem o mínimo de preparo técnico pra preencher as vagas. Temos muitas faculdades com cursos como CC e SI mas que no final nenhum foi preparado pra ser um programador o que cria um gap entre as vagas abertas e os profissionais disponíveis. 

Aqui na empresa, acabamos contratando um estagiário que está no 3 período pra poder treiná-lo. Contratamos apenas pela postura de gostar de estudar e querer aprender. Este pra mim é um perfil muito importante em qualquer candidato, já que trabalhamos com uma plataforma não muito adotada por aqui. 

Minha impressão é que estamos em um apagão profissional. Muita gente querendo entrar no mercado com canudo na mão e vagas que exigem skills que estes não possuem, muitas vezes por falta de interesse ou informação, já que temos muito conteúdo, livros e cursos disponíveis, inclusive muitos gratuitos.

**Comment by DanielLCintra** [comment]
Fui Gerente de Projetos e Lider de Desenvolvimento na ultima empresa que trabalhei, e na ocasião tive que entrevistar vários candidatos para entrar na equipe. 

Eu costumo falar que a empresa ( e eu particularmente) procura pessoas diferenciadas, porque pessoas comuns que sabem o básico (php + javascript) tem aos montes por ai.. Mas pessoas que buscam estudar javascript a fundo (ES6 e etc), estudar frameworks de JS, PHP (orientado a objetos),  frameworks de PHP.. são poucas.. e quando eu tive que contratar isso pesou pra mim.. Porque eu acredito que se pessoa teve a curiosidade e a vontade de aprender mais e não simplesmente ficar só no básico ela não vai ter medo de aprender nada novo, vai sempre encarar dificuldades como desafios e vai estar crescendo cada vez mais. Acho que esse tipo de pessoa que as empresas estão procurando, profissionais que se desafiam todos os dias, e tem sede por conhecimento. 

E pró-atividade, (parece clichê de pessoal de RH) mas não é. Tive experiencias dos dois lados, devs muito pró-ativos que sempre buscavam soluções novas, propunham melhorias, e sempre estavam dando feedback do trabalho que estava sendo realizado, isso faz o trabalho fluir muito mais redondo. tbm tive experiencia com devs que só faziam o que eram mandados e mesmo se não fosse a melhor solução, nem arriscavam dizer que existe algo melhor ou tentar fazer de uma forma diferente, isso é muito ruim.

**Comment by fernahh** [comment]
Onde eu trabalho temos bastante vagas pra desenvolvedores, tanto back-end quanto front-end. As vagas pra back-end tem um ciclo bem menor do que as de front, e geralmente candidatos com um nível de senioridade maior. Acredito que isso seja reflexo do tempo em que essa especialidade existe, visto que aplicações SPA sugiram há mais ou menos ~7 anos.

> quais os skills que estão faltando nas pessoas que estão participando dos seus processos seletivos?

Respondendo diretamente a pergunta, dos processo seletivos que participei aqui dentro, acho que:

- Poucos candidatos tem experiência com SPA, boa parte, como o @bernarddeluna falou, não investiram/investem em sair do jQuery;
- Muitos candidatos, por incrível que pareça, tem ~orgulho~ de não saber CSS (isso é muito triste, sério);
- Uma boa parte usa os frameworks orientando a view, mesmo os frameworks modernos (react, vue, ng2+) terem suas arquiteturas baseadas em componentes. É bem comum eu pegar testes Vue, ou Angular em que existe um arquivo HTML por rota e nesses caras cheio de diretivas no template.
- Existe uma parte considerável de desenvolvedores que espera encontrar um mundo perfeito, onde tudo está há um passo de um `npm install`. Na realidade, esse cenário é difícil de encontrar, inclusive [escrevi sobre isso](https://engineering.contaazul.com/o-santo-graal-da-arquitetura-de-software-n%C3%A3o-existe-7c4cdcc75475). Desenvolvedores precisam resolver problemas reais do mercado, e boa parte desses problemas não se resolvem instalando uma biblioteca. É importante aprender design patterns e o porquê e como funcionam determinadas ferramentas.

**Comment by matheusml** [comment]
Sou um dos _leads_ de JavaScript da [Concrete](www.concrete.com.br) e faço parte da equipe de recrutamento desde 2015.
Já fiz a entrevista de dezenas de candidatos (talvez centenas). E geralmente é isso o que eu menos gosto nos candidatos:

### Técnica
- Não sabe os conceitos mais básicos de JavaScript, HTML e/ou CSS
- Está completamente desatualizado. Alguns casos recentes o candidato disse que Angular 1 era novidade, por exemplo.
- Não ter interesse em estudar novas tecnologias (de certa forma está relacionado ao item anterior)

### Não técnica
- Chegar atrasado na entrevista
- Mentir
- Não saber se comunicar
- Não saber o que quer. As vezes pergunto se o candidato prefere trabalhar com desenvolvimento _web_ ou _mobile_ e o mesmo responde, "tanto faz".
- Não saber porque quer trabalhar na empresa

Espero ter ajudado 👍 👍 👍

**Comment by patrickporto** [comment]
Sou um dos Tech Leaders na Stone e faço parte do processo de recrutamento de candidatos. Tenho um processo de avaliação que envolve um desafio técnico onde também é avaliado o que não é técnico. O que considero pontos negativos:

**Desafio**

* O candidato não tira dúvida. Nos colocamos a disposição para tirar dúvidas porém o mesmo opta por seguir e fazer uma implementação que o mesmo não entende.
* O candidato não sabe interpretar o que foi pedido. Nós sempre estamos dispostos a responder dúvidas, porém muitas questões estão explícitas no enunciado. No mundo real um Desenvolvedor é um ser pensante, não apenas uma máquina de escrever código que espera alguém dizer o que precisa ser feito.
* O candidato entrega estritamente o que foi pedido, sem se preocupar com detalhes intrínsecos. Como o caso de uma API que oferece paginação e o mesmo exibir apenas a primeira página. Quando os requisitos são escritos, existe uma preocupação para que estejam em alto nível, como a solicitação de uma pessoa de negócios.
* A falta de domínio em uma tecnologia não é um problema, porém não conseguir justificar as decisões técnicas é algo que pesa demais.

**Entrevista**

* Dizer que só trabalhou com pessoas incompetentes. Diminuir seus companheiros de trabalho a favor de suas próprias competências é buscar justificar sua própria falta de trabalho em equipe. Se alguém trabalha com você e tu não agrega nada a ela então essa é uma falha enorme no compartilhamento de conhecimento.
* A falta de pontualidade não é encarada como um problema tão grande quanto a falta de compromisso, mesmo que em muitos lugares sejam ambos sinônimos. Chegar atrasado ou até mesmo faltar a entrevista sem avisar com antecedência é uma falta de compromisso considerada grave.
* Mentir desclassifica na hora pois o mínimo que precisamos ter no ambiente de trabalho é confiança uns nos outros.

**Comment by danielschmitz** [comment]
Eu nunca entrevistei ninguém, mas sigo de perto os programadores que possuem dúvidas sobre os meus livros, e infelizmente, na maioria dos casos (salvo exceções) o que os leitores querem é pegar o código fonte e "rodar" na máquina deles. Eles se negam a ter que digitar código fonte, querem copiar e colar, e assim acham que estão aprendendo. 

Falta também um interesse geral nas tecnologias que surgem a cada dia. Um programador que está focado hoje no jquery é um programador morto para mim (como a @bernarddeluna  disse).

**Comment by PauloDanielCarneiro** [comment]
> Além do que citaram acima, o que pecam muito é a análise. Muitos devs estão focados apenas em programar, é não visualizam o motivo daquilo tudo.
Escrever código é "fácil", o difícil é agregar valor com a entrega.
Como citou o @klaygomes, conhecer o negócio da empresa faz toda a diferença.
Tecnologia é "apenas" ferramenta.

Mas se vc for pensar, programar é o trabalho do programador. O analista é quem faz o papel de "agregador". Isso é complicado aqui no Brasil. Não pode negar que sempre querem acumular diversas funções em um cara só.

> Não são devs, são apenas programadores 😄. Desenvolvedor também programa, mas programador não necessariamente desenvolve...

Realmente discordo. Desenvolver, nesse aspecto é programar. Entender do negocio, é analise. São areas distintas. Um é MUITO mais multidisciplinar do que o outro. Ou estou errado?

**Comment by bernardodiasc** [comment]
> Realmente discordo. Desenvolver, nesse aspecto é programar. Entender do negocio, é analise. São areas distintas. Um é MUITO mais multidisciplinar do que o outro. Ou estou errado?

Desenvolver não é programar. Existe uma diversidade imensa de funções de desenvolvimento de software que não são programação.

**Comment by synga** [comment]
Bom, tanto nessa issue quanto na #917 da pra se ver um padrão onde em ambos os lados no que mais se peca é comunicação (isso tirando parte técnica). Da parte da empresa há falta de informações e feedback pelo que a galera reportou e da parte do candidato há uma deficit grande de comunicação, saber se portar, se comunicar, perguntar, se expressar.

Agora tirando um pouco essa discussão do foco atual e puxando o foco pra nós mesmos, será que não ta na hora de nós como devs trabalharmos mais essa parte de comunicação? Realmente tem MUITO dev que tem vergonha/medo de se comunicar, não se se é cultural, mas é fácil perceber quando a pessoa é assim. Ainda que pra nós de SP tem vários eventos, meetups, grupos, fóruns e etc e isso facilita muito na hora de ter essa pratica de conversação.

Quando fui estagiário eramos em 6, uma empresa nos contratou para um programa de treinamento de 3 meses em tecnologias Microsoft. Dessas 6 pessoas 1 menina em questão NÃO conversava, não tirava duvida, não ia atrás e nem nada, ela até sabia se virar e aprendeu bem a programar, mas foi a unica não contratada e simplesmente pelo motivo de não saber trabalhar em grupo, não se comunicar, travava nas daily do SCRUM. Não há uma troca em muita gente de T.I., tem muitos que vivem numa bolha ainda, nisso entra a parte da tecnologia e gente que não se atualiza.

Até hoje quando falo com alguns amigos sobre JavaScript eles dão risada, alguns falam pra eu largar isso e ir estudar C#, Java ou Python, desenvolvimento mobile só existe nativo, front-end é Bootstrap com JQuery e um template de Wordpress, Node não vai pra frente e muitas coisas que ouvia no começo da faculdade e continuam nisso.

Será que não ta na hora das empresas e candidatos começarem a buscar essa troca? Por exemplo, ao invés de uma dinâmica de grupo do tipo "resolvam esse problema" que tal trazer alguns assuntos para discussão entre todos? Ou colocar anúncios de vagas mais amigáveis e bem mais detalhadas? Com mais comunicação provavelmente muitos dos problemas na contratação seriam sanados, a não ser a parte técnica que viria a um prazo um pouco mais longo, mas já seria um avanço enorme.

**Comment by dbuarque** [comment]
Para todo processo de seleção que abrimos aqui na [Getty/IO](https://getty.io) recebemos em torno de 200+ aplicações.

Algumas informações relevantes:

- 40% dos candidatos não chegam nem a serem avaliados por que não leram a descrição da vaga.
- 99% dos candidatos não completam os desafios, muitas vezes nem chegam a começar.
- 90% dos candidados são medianos. Ou seja, não se preocupam em se diferenciar dos demais.

Dicas:
Caso você não seja selecionado(a), melhore seu currículo, melhor seu linkedin, adicione exemplos de código do seu github, seja ativo na comunidade, consiga a certificação do freecodecamp e outras, tente escrever um artigo técnico. Isso vai aumentar absurdamente suas chances de ser selecionado.

Falar inglês também é um ponto de muito diferencial. Para ser um desenvolvedor(a) top, você precisa dominar o inglês perfeitamente.

Prioridade no inglês enquanto estiverem fazendo faculdade. Façam intercambio, vale a pena!

Se o candidato quer ser um profissional top de mercado, ele precisa a estar disposto a agir como um. Se ele tomar 10x mais ações, ele terá 10x mais resultados que os outros.
